### DISCUSSION TITLE
Should some fields be mandatory ?

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#401] [Q&A] [answered]
Good Morning Lab !
I am using sdk 1.6.
Filling notice subtype 9 with mandatory fields only, I get an error during xsd validation, due the lack of "ID" fields.
I am sending a Jpg as sample. Should these fields be mandatory ?

![Information](https://user-images.githubusercontent.com/52211037/229702865-83b8e1ba-9899-444d-b52d-d9926d4aacdc.jpg)

<xsd:complexType name="DocumentReferenceType">
      <xsd:sequence>
         <xsd:element ref="ext:UBLExtensions" minOccurs="0" maxOccurs="1"/>
         <xsd:element ref="cbc:ID" minOccurs="1" maxOccurs="1"/>
***********************************************************************************
         <xsd:element ref="cac:IssuerParty" minOccurs="0" maxOccurs="1"/>
         <xsd:element ref="cac:ResultOfVerification" minOccurs="0" maxOccurs="1"/>
      </xsd:sequence>
Thanks and Regards !
Renato

**Comment by YvesJo** [comment, marked as answer]
Hi Renato,
These IDs are mandatory for any specified document reference. Dedicated BTs OPT-111, OPT-112 and OPT-113 and their associated fields have been defined for that purpose. Rules on these fields may need a review on our side.
